diff --git a/vmmon-only/Makefile.kernel b/vmmon-only/Makefile.kernel
index a39e099..0bbf1ca 100644
--- a/vmmon-only/Makefile.kernel
+++ b/vmmon-only/Makefile.kernel
@@ -21,6 +21,7 @@ CC_OPTS += -DVMMON -DVMCORE
 
 INCLUDE := -I$(SRCROOT)/include -I$(SRCROOT)/include/x86 -I$(SRCROOT)/common -I$(SRCROOT)/linux
 EXTRA_CFLAGS := $(CC_OPTS) $(INCLUDE)
+ccflags-y := $(CC_OPTS) $(INCLUDE)
 
 obj-m += $(DRIVER).o
 
diff --git a/vmmon-only/common/task.c b/vmmon-only/common/task.c
index 5d34cf4..f375206 100644
--- a/vmmon-only/common/task.c
+++ b/vmmon-only/common/task.c
@@ -544,7 +544,7 @@ Task_Terminate(void)
  *-----------------------------------------------------------------------------
  */
 
-Selector
+static Selector
 TaskGetFlatWriteableDataSegment(void)
 {
    DTR hostGDTR;
diff --git a/vmmon-only/common/vmx86.c b/vmmon-only/common/vmx86.c
index 86e52fb..cf64b12 100644
--- a/vmmon-only/common/vmx86.c
+++ b/vmmon-only/common/vmx86.c
@@ -49,7 +49,7 @@
 #include "x86svm.h"
 #include "x86cpuid_asm.h"
 #if defined(__linux__)
-#include <asm/timex.h>
+#include <linux/timex.h>
 #endif
 #include "perfctr.h"
 #include "x86vtinstr.h"
@@ -696,7 +696,7 @@ Vmx86FreeCrossPages(VMDriver *vm)
  *-----------------------------------------------------------------------------
  */
 
-void
+static void
 Vmx86FreeVMDriver(VMDriver *vm)
 {
    Vmx86_Free(vm->ptRootMpns);
@@ -729,7 +729,7 @@ Vmx86FreeVMDriver(VMDriver *vm)
  *-----------------------------------------------------------------------------
  */
 
-VMDriver *
+static VMDriver *
 Vmx86AllocVMDriver(uint32 numVCPUs)
 {
    VMDriver *vm = Vmx86_Calloc(1, sizeof *vm, TRUE);
diff --git a/vmmon-only/include/x86cpuid.h b/vmmon-only/include/x86cpuid.h
index 1fe6fcd..2996ae4 100644
--- a/vmmon-only/include/x86cpuid.h
+++ b/vmmon-only/include/x86cpuid.h
@@ -92,7 +92,7 @@ typedef struct CPUIDQuery {
    uint32 eax;                // IN
    uint32 ecx;                // IN
    uint32 numLogicalCPUs;     // IN/OUT
-   CPUIDReply logicalCPUs[0]; // OUT
+   CPUIDReply logicalCPUs[]; // OUT
 } CPUIDQuery;
 #pragma pack(pop)
 #endif
diff --git a/vmmon-only/include/x86msr.h b/vmmon-only/include/x86msr.h
index 189c982..bc0e6ef 100644
--- a/vmmon-only/include/x86msr.h
+++ b/vmmon-only/include/x86msr.h
@@ -67,7 +67,7 @@ typedef struct MSRReply {
 typedef struct MSRQuery {
    uint32 msrNum;           // IN
    uint32 numLogicalCPUs;   // IN/OUT
-   MSRReply logicalCPUs[0]; // OUT
+   MSRReply logicalCPUs[]; // OUT
 } MSRQuery;
 #pragma pack(pop)
 
diff --git a/vmmon-only/linux/driver.c b/vmmon-only/linux/driver.c
index 3d8b722..6aaacd9 100644
--- a/vmmon-only/linux/driver.c
+++ b/vmmon-only/linux/driver.c
@@ -267,7 +267,7 @@ LinuxDriverInitTSCkHz(void)
  *----------------------------------------------------------------------
  */
 
-int
+static int
 LinuxDriverInit(void)
 {
    int retval;
@@ -335,7 +335,7 @@ LinuxDriverInit(void)
  *----------------------------------------------------------------------
  */
 
-void
+static void
 LinuxDriverExit(void)
 {
    /*
@@ -344,8 +344,11 @@ LinuxDriverExit(void)
    misc_deregister(&vmmon_miscdev);
 
    Log("Module %s: unloaded\n", vmmon_miscdev.name);
-
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 15, 0)
    del_timer_sync(&tscTimer);
+#else
+   timer_delete_sync(&tscTimer);
+#endif
 
    Vmx86_CleanupHVIOBitmap();
    Task_Terminate();
diff --git a/vmmon-only/linux/hostif.c b/vmmon-only/linux/hostif.c
index b9e3d30..2c49ec0 100644
--- a/vmmon-only/linux/hostif.c
+++ b/vmmon-only/linux/hostif.c
@@ -1995,7 +1995,11 @@ HostIF_InitUptime(void)
 void
 HostIF_CleanupUptime(void)
 {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 15, 0)
    del_timer_sync(&uptimeState.timer);
+#else
+   timer_delete_sync(&uptimeState.timer);
+#endif
 }
 
 
@@ -2922,7 +2926,7 @@ HostIF_CallOnEachCPU(void (*func)(void*), // IN: function to call
  *-----------------------------------------------------------------------------
  */
 
-Bool
+static Bool
 HostIFCheckTrackedMPN(VMDriver *vm, // IN: The VM instance
                       MPN mpn)      // IN: The MPN
 {
@@ -3042,7 +3046,7 @@ HostIF_ReadPhysical(VMDriver *vm,      // IN: The VM instance
  *----------------------------------------------------------------------
  */
 
-int
+static int
 HostIFWritePhysicalWork(MA ma,             // MA to be written to
                         VA64 addr,         // src data to write
                         Bool kernelBuffer, // is the buffer in kernel space?
@@ -3201,7 +3205,7 @@ HostIF_GetCurrentPCPU(void)
  *----------------------------------------------------------------------
  */
 
-int
+static int
 HostIFStartTimer(Bool rateChanged,  //IN: Did rate change?
                  unsigned int rate) //IN: current clock rate
 {
@@ -3408,8 +3412,11 @@ HostIF_SafeRDMSR(unsigned int msr,   // IN
 {
    int err;
    u64 v;
-
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 16, 0)
    err = rdmsrl_safe(msr, &v);
+#else
+   err = rdmsrq_safe(msr, &v);
+#endif
    *val = (err == 0) ? v : 0;  // Linux corrupts 'v' on error
 
    return err;
diff --git a/vmnet-only/bridge.c b/vmnet-only/bridge.c
index bc4727c..be8140e 100644
--- a/vmnet-only/bridge.c
+++ b/vmnet-only/bridge.c
@@ -1411,7 +1411,7 @@ VNetBridgeComputeHeaderPos(struct sk_buff *skb) // IN: buffer to examine
  *----------------------------------------------------------------------
  */
 
-void
+static void
 VNetBridgeSendLargePacket(struct sk_buff *skb,        // IN: packet to split
                           VNetBridge *bridge)         // IN: bridge
 {
diff --git a/vmnet-only/driver.c b/vmnet-only/driver.c
index f314ff2..4cf1335 100644
--- a/vmnet-only/driver.c
+++ b/vmnet-only/driver.c
@@ -296,7 +296,7 @@ VNetRemovePortFromList(const VNetPort *port) // IN: port to remove from list
  */
 
 int
-vmnet_init_module(void)
+static vmnet_init_module(void)
 {
    int retval;
 
@@ -374,7 +374,7 @@ err_proto:
  *----------------------------------------------------------------------
  */
 
-void
+static void
 vmnet_cleanup_module(void)
 {
    unregister_chrdev(VNET_MAJOR_NUMBER, "vmnet");
diff --git a/vmnet-only/vmnetInt.h b/vmnet-only/vmnetInt.h
index b49c399..f1d8e9e 100644
--- a/vmnet-only/vmnetInt.h
+++ b/vmnet-only/vmnetInt.h
@@ -41,9 +41,13 @@
     compat_skb_set_network_header(skb, sizeof (struct ethhdr)),  \
     dev_queue_xmit(skb)                                   \
   )
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 14, 0)
 #define dev_lock_list()    read_lock(&dev_base_lock)
 #define dev_unlock_list()  read_unlock(&dev_base_lock)
-
+#else
+#define dev_lock_list()    rcu_read_lock()
+#define dev_unlock_list()  rcu_read_unlock()
+#endif
 
 extern struct proto vmnet_proto;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 2, 0) || defined(sk_net_refcnt)
diff --git a/vmnet-only/vnetInt.h b/vmnet-only/vnetInt.h
index f6a90a1..c9e8953 100644
--- a/vmnet-only/vnetInt.h
+++ b/vmnet-only/vnetInt.h
@@ -218,7 +218,12 @@ extern int VNetProc_Init(void);
 
 extern void VNetProc_Cleanup(void);
 
-
+// int VNetNetIf_Create(char *devName, VNetPort **ret, int hubNum);
+// int VNetUserIf_Create(VNetPort **ret);
+// int VNetBridge_Create(const char *devName, uint32 flags, VNetJack *hubJack,
+// 		      VNetPort **ret);
+// int VNetUserListener_Create(uint32 classMask, VNetJack *hubJack,
+//			    VNetPort **port);
 /*
  *----------------------------------------------------------------------
  *
